File: commands/wipe.js
import chalk from 'chalk'
import { request } from '../libraries/request.js'
import { confirm } from '../libraries/prompt.js'

export async function wipe() {
  const test = await confirm(
    chalk.red('WARNING: This will remove ALL users from the broker. Continue?')
  )

  if (!test) return
  try {
    const res = await request('wipe', {})

    if (res.success) {
      console.log(chalk.green(res.success))
    } else {
      console.log(chalk.red(res.error))
    }
  } catch (err) {
    console.log(chalk.red('Problem connecting to server.'))
    console.log(chalk.red(err))
  }
}

File: cmd/wipe.go
package cmd

import (
	"fmt"

	"github.com/nate/credential-broker/client"
	"github.com/nate/credential-broker/prompt"
	"github.com/spf13/cobra"
)

func runWipe(cmd *cobra.Command, args []string) error {
	confirmed, err := prompt.Confirm("This will wipe ALL users from the broker. Are you sure?")
	if err != nil {
		return err
	}
	if !confirmed {
		return nil
	}

	confirmed2, err := prompt.Confirm("Are you absolutely certain? This cannot be undone.")
	if err != nil {
		return err
	}
	if !confirmed2 {
		return nil
	}

	resp, err := client.Request("wipe", "")
	if err != nil {
		return fmt.Errorf("wipe failed: %w", err)
	}

	printResponse(resp)
	return nil
}
